Makefile

```diff
@@ -0,0 +1,102 @@
+
+update: stack.yml environment.yml
+	@touch .deploy.$(STACKNAME)
+	openstack stack update -t $< -e environment.yml $(STACK_PARAMS) $(STACKNAME)
+
+# Cleanup
+clean:
+	openstack stack delete -y $(STACKNAME)
+	@rm -f .deploy.$(STACKNAME) .MANAGER_ADDRESS.$(STACKNAME)
+	rm -f ~/.ssh/id_rsa.$(STACKNAME)
+
+clean-wait:
+	openstack stack delete -y --wait $(STACKNAME)
+	@rm -f .deploy.$(STACKNAME) .MANAGER_ADDRESS.$(STACKNAME)
+	rm -f ~/.ssh/id_rsa.$(STACKNAME)
+
+# To recover from stale ssh-key and MANAGER_ADDRESS
+reset:
+	@rm .deploy.$(STACKNAME)
+
+# Watch the creation of the stack
+watch: .deploy.$(STACKNAME)
+	MGR_ADR=""; STAT=""; while true; do\
+		date; openstack stack list; \
+		SRV=$$(openstack server list -f value -c "Name" -c "Status" | grep testbed-manager | cut -d ' ' -f2); \
+		openstack server list; \
+		if test -z "$$MGR_ADR" -a "$$SRV" = "ACTIVE"; then \
+			openstack stack output show $(STACKNAME) private_key -f value -c output_value > ~/.ssh/id_rsa.$(STACKNAME); \
+			chmod 0600 ~/.ssh/id_rsa.$(STACKNAME); \
+			MGR_ADR=$$(openstack stack output show $(STACKNAME) manager_address -f value -c output_value); \
+			echo "MANAGER_ADDRESS=$$MGR_ADR" > .MANAGER_ADDRESS.$(STACKNAME); \
+		fi; \
+		if test -n "$$MGR_ADR"; then ssh -o StrictHostKeyChecking=no -i ~/.ssh/id_rsa.$(STACKNAME) ubuntu@$$MGR_ADR "sudo grep PLAY /var/log/cloud-init-output.log | grep -v 'PLAY \(\[\(Group hosts\|Gather facts\)\|RECAP\)' | tail -n3; sudo tail -n12 /var/log/cloud-init-output.log"; fi; \
+		STAT=$$(openstack stack list -f value -c "Stack Name" -c "Stack Status" | grep $(STACKNAME) | cut -d' ' -f2); \
+		if test "$$STAT" == "CREATE_COMPLETE"; then break; fi; \
+		if test "$$STAT" == "CREATE_FAILED"; then openstack stack show $(STACKNAME) -f value -c "stack_status_reason"; break; fi; \
+		echo; sleep 30; \
+	done
+
+# Look for stack
+.deploy.$(STACKNAME):
+	STAT=$$(openstack stack list -f value -c "Stack Name" -c "Stack Status" | grep $(STACKNAME) | cut -d' ' -f2); \
+	if test -n "$$STAT"; then touch .deploy.$(STACKNAME); else echo "use make deploy or deploy-infra or ...."; exit 1; fi
+
+# Get output
+~/.ssh/id_rsa.$(STACKNAME): .deploy.$(STACKNAME)
+	openstack stack output show $(STACKNAME) private_key -f value -c output_value > $@
+	chmod 0600 $@
+
+.MANAGER_ADDRESS.$(STACKNAME): .deploy.$(STACKNAME)
+	@MANAGER_ADDRESS=$$(openstack stack output show $(STACKNAME) manager_address -f value -c output_value); \
+	echo "MANAGER_ADDRESS=$$MANAGER_ADDRESS" > $@; \
+	echo $$MANAGER_ADDRESS
+
+# Convenience: sshuttle invocation
+sshuttle: ~/.ssh/id_rsa.$(STACKNAME) .MANAGER_ADDRESS.$(STACKNAME)
+	source ./.MANAGER_ADDRESS.$(STACKNAME); \
+	sshuttle --ssh-cmd "ssh -i $<" -r dragon@$$MANAGER_ADDRESS 192.168.40.0/24 192.168.50.0/24 192.168.90.0/24
+
+ssh_manager: ~/.ssh/id_rsa.$(STACKNAME) .MANAGER_ADDRESS.$(STACKNAME)
```

Comment:
I think ssh makes more sense here. You can only connect to the manager via SSH.

Reply:
```
STAT=$(openstack stack list -f value -c "Stack Name" -c "Stack Status" | grep testbed | cut -d' ' -f2); \
	if test -n "$STAT"; then touch .deploy.testbed; else echo "use make deploy or deploy-infra or ...."; exit 1; fi
Missing value auth-url required for auth plugin password
use make deploy or deploy-infra or ....
make: *** [.deploy.testbed] Error 1
```

Reply:
```export OS_CLOUD=testbed``` required. 

I think it makes sense to introduce --os-cloud $OS_CLOUD and set OS_CLOUD to testbed by default.

Reply:
make ssh is fine with me.
I don't want to force the usage of --os-cloud, so I'll have to do a bit of magic to not break parameterless openstack invocation either (both classic OS_ style config or exported OS_CLOUD var).

Reply:
OK, so all cases work now.
1. Old config (OS_PASSWORD)
2. New config with OS_CLOUD exported
3. New config without OS_CLOUD, passing --os-cloud=testbed
4. New config without OS_CLOUD, make param OS_CLOUD=override, passing --os-cloud=override.